replace tf::MessageFilter by tf2_ros::MessageFilter

christian-rauch opened the pull request:
Replace `tf::MessageFilter` by `tf2_ros::MessageFilter` for ROS melodic.

k-okada commented:
Thanks for fix!
BTW, did you check if this is compilable? We did not check melodic build on our Travis, but at least, it seems failing on kinetic. Is it expected behavior? 
```
_______________________________________________________________________________
Errors << jsk_pcl_ros:make /workspace/ros/ws_jsk_recognition/logs/jsk_pcl_ros/build.make.000.log
/usr/local/lib/python2.7/dist-packages/requests/__init__.py:83: RequestsDependencyWarning: Old version of cryptography ([1, 2, 3]) may cause slowdown.
  warnings.warn(warning, RequestsDependencyWarning)
/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_pcl_ros/src/pointcloud_moveit_filter.cpp: In member function 'virtual bool jsk_pcl_ros::PointCloudMoveitFilter::initialize()':
/workspace/ros/ws_jsk_recognition/src/jsk_recognition/jsk_pcl_ros/src/pointcloud_moveit_filter.cpp:97:9: error: no match for 'operator=' (operand types are 'std::shared_ptr<tf2_ros::Buffer>' and 'const boost::shared_ptr<tf::Transformer>')
     tf_ = monitor_->getTFClient();
         ^
```

furushchev commented:
@christian-rauch @k-okada Just from curiosity, how does this fix change the behavior of the node?
If  I understand correctly, `tf` has exactly the same functionality except the concept of `BufferClient` since it uses `tf2` internally for resolving transformation tree.

christian-rauch commented:
This PR is only for fixing melodic build, it will break kinetic builds.
The newer [`OccupancyMapMonitor`](http://docs.ros.org/api/moveit_ros_perception/html/classoccupancy__map__monitor_1_1OccupancyMapMonitor.html#a15c47655e83db3df7cc14fa10b82f55c) returns `tf2_ros::Buffer` for `monitor_->getTFClient();`. `tf2_ros` should be supported in kinetic, but the older `getTFClient()` returns `tf::Transformer`.

If you create a new `melodic-devel` branch, I will rebase on it.

christian-rauch commented:
Can someone (@k-okada, @furushchev) trigger the kinetic CI build?
It failed because there is not enough space on the device:
`E: You don't have enough free space in /var/cache/apt/archives/.`

christian-rauch commented:
I am still getting `You don't have enough free space in /var/cache/apt/archives/.`. Can someone fix the CI?

christian-rauch commented:
I used the `ROS_VERSION_MINIMUM` macro to switch between the `tf` and `tf2_ros` versions of the transformer and message filter.

christian-rauch commented:
The CI times out after I added a new travis configuration for an optional melodic build.
Apart from this, this PR is read to merge (@k-okada).

k-okada commented:
Thanks, but I’d like to pass melodic test on Travis, before merging this.

2018年11月22日(木) 21:29 Naoki Mizuno <notifications@github.com>:

> FYI, I've also confirmed on my melodic that this PR compiles.
>
> —
> You are receiving this because you were mentioned.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/2335#issuecomment-441015926>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3MlXfEtpjJUCuu7EEjk8_MhgpvIXks5uxpihgaJpZM4W9cb6>
> .
>
-- 
--
◉ Kei Okada

k-okada commented:
`jsk_recognition` depends on `robot_self_flter` and `octmap_mapping` which are not released on melodic. So how did @christian-rauch @naoki-mizuno compiled on melodic? Did you install theses packages from source?
https://github.com/pr2/robot_self_filter/issues
https://github.com/OctoMap/octomap_mapping/issues/49

I think we need volunteer for package maintainer, if you could not take full responsibility for package maintainer, you could just take care of release for newer distro as a orphaned package maintainer ( 
http://wiki.ros.org/OrphanedPackage)

To start with, I recommend to ask for repository access (
https://github.com/ros-visualization/view_controller_msgs/issues/3#issuecomment-416849983)

christian-rauch commented:
I am using the `octomap_mapping` (branch `kinetic-devel`) source package in the workspace.

`robot_self_filter` is not required to compile `jsk_pcl_ros`:
```
CMake Warning (dev) at CMakeLists.txt:392 (message):
  robot_self_filter is NOT found.  some nodelet depending on
  robot_self_filter will not be compiled.
```

This PR enables compiling `jsk_recognition` on `melodic`. It's for the benefit for everyone that wants to use the RViz plugins on ROS melodic. I am not intending to release it, since I am not the maintainer of `jsk_recognition` (and I do not intent to become the maintainer).

k-okada commented:
oh, I mean someone has to release `robot_self_filter` and `octmap_mapping`,
and I volunteered for `robot_self_filter`
https://github.com/ros/rosdistro/pull/19525
--
◉ Kei Okada



2018年11月23日(金) 20:44 Christian Rauch <notifications@github.com>:

> I am using the octomap_mapping (branch kinetic-devel) source package in
> the workspace.
>
> robot_self_filter is not required to compile jsk_pcl_ros:
>
> CMake Warning (dev) at CMakeLists.txt:392 (message):
>   robot_self_filter is NOT found.  some nodelet depending on
>   robot_self_filter will not be compiled.
>
> This PR enables compiling jsk_recognition on melodic. It's for the
> benefit for everyone that wants to use the RViz plugins on ROS melodic. I
> am not intending to release it, since I am not the maintainer of
> jsk_recognition (and I do not intent to become the maintainer).
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jsk-ros-pkg/jsk_recognition/pull/2335#issuecomment-441220735>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAeG3JT8N8Gb9rd1bBuuaoMOUguxxWyrks5ux9-fgaJpZM4W9cb6>
> .
>

christian-rauch commented:
Does this mean that this PR is on halt until someone decides to volunteer for maintaining `octomap_mapping`?
At the moment, this PR allows using `jsk_recognition` on `melodic`. Without, everyone needs to reimplement the changes for `melodic` by themselves.

wxmerkt commented:
I have made a release request for `octomap_mapping` into `melodic`, cf. https://github.com/ros/rosdistro/pull/19985. Are there other outstanding issues to get this PR merged and released for melodic?

christian-rauch commented:
Thanks @wxmerkt for maintaining the `octomap_mapping` package.
I rebased on master and also changed the signature of `PointCloudMoveitFilter::getShapeTransform` to match the Eigen transformation type after API breakage in MoveIt!.

christian-rauch commented:
This builds now on `indigo` and `kinetic` but is still missing the `jsk_rviz_plugins` (from `jsk_visualization`) for `melodic`.
@k-okada Would you consider merging this or releasing `jsk_visualization` and adding melodic to the build targets that need to pass?

k-okada commented:
@christian-rauch thanks for contribution, your patches has been merged as a part of #2355
